__bass.py

```diff
@@ -10,24 +10,24 @@
 from __future__ import print_function
 
 import json
-import os
 import subprocess
 import sys
 import tempfile
 
 
-def gen_script():
-    fd, name = tempfile.mkstemp()
+BASH = 'bash'
+
 
+def gen_script():
     divider = '-__-__-__bass___-env-output-__bass_-__-__-__-__'
 
-    old_env = os.popen('/bin/bash -c "env"', 'r').read().splitlines()
+    args = [BASH, '-c', 'env']
+    output = subprocess.check_output(args, universal_newlines=True)
+    old_env = output.splitlines()
 
     command = '{}; echo "{}"; env'.format(' '.join(sys.argv[1:]), divider)
-    output = subprocess.check_output(['bash', '-c', command])
-    if bytes != str and isinstance(output, bytes):
-        # on python 3
-        output = str(output, 'utf-8')
+    args = [BASH, '-c', command]
+    output = subprocess.check_output(args, universal_newlines=True)
```

Comment:
`output` will be of `bytes` type on Python 3. That would break the next line IIUC. Could you add back line 28-30?